### DISCUSSION TITLE
Missing Passport Class

### DISCUSSION THREAD

**Discussion by ivasniel** [netsuitephp/netsuite-php#143] [Q&A] [answered]
Im getting this error every time that i try to send a request to Netsuite. Any idea why the package does not include this class?

![image](https://user-images.githubusercontent.com/19581318/213305040-be3ada6e-602e-4398-9c0d-f7d55c4ce955.png)

{"data":[],"errors":{"api":["Class \"NetSuite\\Classes\\Passport\" not found"]},"message":"Error retrieving record","status":"fail","success":false,"errorCode":0}

**Comment by jrebs** [comment, marked as answer]
NetSuite dropped support for that authentication style in 2020 and as of 2021.0 release it was no longer a class provided with their toolkit. You could use an older version of the repository to still have that class available, but afaik it won't work because I think NetSuite doesn't support it any longer.

**Reply by ivasniel** [reply]
Thanks for your quick response, im not using it, but it's used inside of NetSuiteClient.php.

**Reply by jrebs** [reply]
We'll have to remove that code path in a future version, but if you're hitting that part of the code, then you haven't configured the NetSuiteClient appropriately. NetSuite only supports the token method now: https://github.com/netsuitephp/netsuite-php/blob/master/src/NetSuiteClient.php#L188

**Reply by ivasniel** [reply]
Ok, i will review it, thanks

**Comment by ivasniel** [comment]
Yes, you were right,  the configuration was wrong, now it's working. Thanks for everything.
